client/src/GameState.ts

```diff
@@ -138,6 +215,158 @@ export class GameState {
         this.emitPlayerMove();
     }
 
+    public updateLineClearing() {
+        this.monominoesToDraw = this.monominoesToDraw.concat(
+            this.lineCheckSequence
+                .map((task) => {
+                    const linesToClear = this.scanLinesToClear(task);
+                    if (linesToClear.length === 0) {
+                        return [];
+                    }
+                    // create a map to record by how much we should fall/offset each lines after removing
+                    const linesToFall = this.prepareLinesToFall(
+                        task,
+                        linesToClear
+                    );
+                    this.removeLines(task, linesToClear);
+                    const monominoesToDraw = this.fallLines(task, linesToFall);
+
+                    // TODO scoring
+
+                    return monominoesToDraw;
+                })
+                .reduce((total, curr) => [...total, ...curr])
+        );
+    }
```

Comment:
Getting into extremely nitpicky territory so maybe ignore this lol, but I would store the new monominoes to draw in a var and then do the concat after just to avoid nesting and running into formatter line wrapping, especially with the scoring code that may be there as well.

Reply:
I agree that this nesting does not read very well. But let's leave it here and see what the scoring actually looks like and do some refactor from there if needed.